inst/include/Rcpp/platform/compiler.h

```diff
@@ -57,10 +44,11 @@
         #define IS_GCC_460_OR_LATER
     #endif
     // Inform users of dated compiler (e.g. <= RHEL6 )
-    #if GCC_VERSION <= 40602
-        #pragma message ("WARNING: GCC Compiler Version <= 4.6.2")
-        #pragma message ("WARNING: The compiler being used is both OUTDATED and INCOMPLETE in terms of C++ standards;")
-        #pragma message ("WARNING: if something does not work, you should upgrade your compiler!")
+    #if GCC_VERSION <= 40602 && !defined(RCPP_GCC_COMPILER_WARNING_SILENT)
+         // Ensure not fake GCC flag set by clang or intel.
+         #if !(defined(__clang__) || defined(__INTEL_COMPILER))
```

Comment:
Yes that is the current discussion but I held off on merging as I wanted to hear from you.